src/component/environments/EnvironmentActionCell/EnvironmentActionCell.tsx

```diff
@@ -0,0 +1,191 @@
+import { ActionCell } from 'component/common/Table/cells/ActionCell/ActionCell';
+import {
+    DELETE_ENVIRONMENT,
+    UPDATE_ENVIRONMENT,
+} from 'component/providers/AccessProvider/permissions';
+import {
+    Edit,
+    Delete,
+    OfflineBolt,
+    DragIndicator,
+    PowerSettingsNew,
+} from '@mui/icons-material';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { IconButton, Tooltip } from '@mui/material';
+import { useNavigate } from 'react-router-dom';
+import AccessContext from 'contexts/AccessContext';
+import { useContext, useState } from 'react';
+import { IEnvironment } from 'interfaces/environments';
+import { formatUnknownError } from 'utils/formatUnknownError';
+import EnvironmentToggleConfirm from '../EnvironmentToggleConfirm/EnvironmentToggleConfirm';
+import EnvironmentDeleteConfirm from '../EnvironmentDeleteConfirm/EnvironmentDeleteConfirm';
+import useEnvironmentApi from 'hooks/api/actions/useEnvironmentApi/useEnvironmentApi';
+import useProjectRolePermissions from 'hooks/api/getters/useProjectRolePermissions/useProjectRolePermissions';
+import { useEnvironments } from 'hooks/api/getters/useEnvironments/useEnvironments';
+import useToast from 'hooks/useToast';
+import { useId } from 'hooks/useId';
+import { useSearchHighlightContext } from 'component/common/Table/SearchHighlightContext/SearchHighlightContext';
+
+interface IEnvironmentTableActionsProps {
+    environment: IEnvironment;
+}
+
+export const EnvironmentActionCell = ({
+    environment,
+}: IEnvironmentTableActionsProps) => {
+    const navigate = useNavigate();
+    const { hasAccess } = useContext(AccessContext);
+    const updatePermission = hasAccess(UPDATE_ENVIRONMENT);
+    const search = useSearchHighlightContext();
+
+    const { setToastApiError, setToastData } = useToast();
+    const { refetchEnvironments } = useEnvironments();
+    const { refetch: refetchPermissions } = useProjectRolePermissions();
+    const { deleteEnvironment, toggleEnvironmentOn, toggleEnvironmentOff } =
+        useEnvironmentApi();
+
+    const [deleteModal, setDeleteModal] = useState(false);
+    const [toggleModal, setToggleModal] = useState(false);
+    const [confirmName, setConfirmName] = useState('');
+
+    const handleDeleteEnvironment = async () => {
+        try {
+            await deleteEnvironment(environment.name);
+            refetchPermissions();
```

Comment:
Just checking here, based on the name this looks like it's a promise. Is this intentionally not awaited?

Reply:
These `refetch` functions do not return promises at the moment, but they probably should. Something for later.

Reply:
Makes sense!

Reply:
Makes sense!